core-api/src/main/java/com/optimizely/ab/config/ProjectConfig.java

```diff
@@ -140,6 +145,7 @@ public ProjectConfig(String accountId, String projectId, String version, String
     // v4 constructor
     public ProjectConfig(String accountId,
                          boolean anonymizeIP,
+                         Boolean botFiltering,
```

Comment:
Why is this `Boolean` and not `boolean` like `anonymizeIP` or vice versa?

Reply:
Because  Boolean is nullable and boolean is not.

Reply:
The interesting thing about what you are saying here is that anonymizeIP is also optional based on datafile type (for instance v2 does not contain anonymizeIP).

Reply:
anonymizeIp default value is false and we are not checking in our sdk if its null it's just getting passed false if not set, but for botfiltering we have three values which are True, False and Null, if botFiltering value is null than we don't add its attribute in list.

Reply:
The same holds true for `anonymizeIP` as it can be null in an older datafile.